outline/shadowsocks/client.go

```diff
@@ -28,15 +28,19 @@ import (
 	"github.com/Jigsaw-Code/outline-go-tun2socks/outline"
 	"github.com/Jigsaw-Code/outline-go-tun2socks/outline/connectivity"
 	"github.com/Jigsaw-Code/outline-go-tun2socks/outline/internal/utf8"
-	"github.com/Jigsaw-Code/outline-go-tun2socks/outline/neterrors"
 	"github.com/Jigsaw-Code/outline-internal-sdk/transport"
 	"github.com/Jigsaw-Code/outline-internal-sdk/transport/shadowsocks"
 	"github.com/Jigsaw-Code/outline-internal-sdk/transport/shadowsocks/client"
 	"github.com/eycorsican/go-tun2socks/common/log"
 )
 
 // A client object that can be used to connect to a remote Shadowsocks proxy.
-type Client = outline.Client
+//
+// Somehow redundant because from it seems gomobile only recognizes a struct,
+// but not a type alias (i.e. type Client = outline.Client).
+type Client struct {
```

Comment:
Should we add the `outline` package to the go mobile call? That should fix this issue and it will be cleaner because it's decoupled from "shadowsocks".

Reply:
We can export `outline` package, this will change the signature if we are defining type alias `type Client = outline.Client` (The `Client` class is changed from `shadowsocks.Client` to `outline.Client`, check the following JNI interface).

<img width="641" alt="image" src="https://user-images.githubusercontent.com/93548144/232596149-3a2465d9-b9db-458b-822b-f8fcbad63df0.png">

Anyway, we already introduced a breaking change in the signature (Now [`OutlineTunnel` is an alias of `Tunnel`](https://github.com/Jigsaw-Code/outline-go-tun2socks/blob/master/outline/tun2socks/tunnel.go#L43), so the generated JNI code will always use `Tunnel` instead of the old `OutlineTunnel`). We just need to decide which way is better:

* Redefining the exported package and update the client
* Creating duplicated types for backward compatibility